src/main/java/com/basho/riak/client/core/query/search/YokozunaIndex.java

```diff
@@ -24,7 +24,8 @@
 {
     private final String name;
     private final String schema;
-    
+    private volatile Integer nVal;
```

Comment:
So this protects against the case where two threads have access to the same `YokozunaIndex` object and one calls `withNVal` at the same time as the other calls `getNVal`.